jasperreports/src/net/sf/jasperreports/engine/fill/JRFillTextElement.java

```diff
@@ -1136,7 +1136,7 @@ protected String processMarkupText(String text)
 	{
 		text = JRStringUtil.replaceCRwithLF(text);
 		
-		if (text != null)
+		if (text != null && text.contains("<"))
```

Comment:
RTF markup does not use the < character.

Reply:
Can we extract that to a method that covers all markup variants and not only html ?

Reply:
The markup processor objects are cached, so not much time is spent accessing them. The markup processor implementation could be optimized so that it gives up parsing early, if conditions are not met.
The test for the < character could be in the HTML markup processor.

Reply:
https://github.com/TIBCOSoftware/jasperreports/commit/1190ff094dcec0d90d5f229d704246a9e08717aa